### PR TITLE
MLE-13695 : Add Container Runtime Detection

### PR DESCRIPTION
**Pull request by rwinieski**
#### Checklist: 

-  ##### Owner:

- [X] JIRA_ID as part of branch/PR name
- [ ] Rebase the branch with upstream
- [ ] Squashed all commits into a single commit
- [ ] Added Tests
  
- ##### Reviewer:

- [ ] Reviewed Tests
- [ ] Added to Release Wiki/Jira

### CODE REVIEW

**Review by vitalykorolev — CHANGES_REQUESTED** [review]
This PR contains some of the UBI/rootless changes

File: README.md [hunk 1]
```diff
@@ -1,73 +1,78 @@
...
+To create an initialized MarkLogic Server, pass in the environment variables MARKLOGIC_ADMIN_USERNAME and MARKLOGIC_ADMIN_PASSWORD, and replace {insert admin username}/{insert admin password} with actual values for admin credentials. Use the optional environment variable MARKLOGIC_WALLET_PASSWORD and REALM to set the wallet password and authentication realm of the admin user. If not provided, the wallet-password will default to the value set for admin-password and realm will be set to public. Optionally, you can pass license information in `{insert license}`/`{insert licensee}` to apply your MarkLogic license. To do this, run this this command:
 
-```
+```bash
 $ docker run -d -it -p 8000:8000 -p 8001:8001 -p 8002:8002 \
```
Comment (vitalykorolev) [inline README.md:76]: Are we removing all $ ?

File: README.md [hunk 2]
```diff
@@ -206,43 +214,49 @@ The following environment variables are only useful when building and extending
...
 
 The credentials for the admin user are configured using Docker secrets, and are stored in `mldb_admin_username.txt`, `mldb_admin_password.txt`, and `mldb_wallet_password.txt` files.
 
-## Single node MarkLogic Server on a single VM
+### Single node MarkLogic Server on a single VM
+
 Single node configurations are used primarily on a development machine with a single user.
 
-Create these files on your host machine: `marklogic-centos.yaml`, `mldb_admin_username.txt`, `mldb_admin_password.txt`, and `mldb_wallet_password.txt`. Run the example Docker commands from the same directory that the files were created.
+Create these files on your host machine: `marklogic-single-node.yaml`, `mldb_admin_username.txt`, `mldb_admin_password.txt`, and `mldb_wallet_password.txt`. Run the example Docker commands from the same directory that the files were created.
```
Comment (vitalykorolev) [inline README.md:255]: This is UBI/rootless related change

Comment (vitalykorolev) [reply README.md:255]: And I see more of these below..

File: README.md [hunk 3]
```diff
@@ -844,174 +908,201 @@ $ docker run -d -it -p 7000:8000 -p 7001:8001 -p 7002:8002 \
...
+
+## Container Runtime Detection
+
+Since MarkLogic 11.2, MarkLogic is able to detect on which container runtime it is running on.
+
+### Docker Engine
+
+When running on Docker Engine the following entry will show up in the ErrorLogs.txt:
+
+`2024-03-15 08:27:36.136 Info: MarkLogic Server is running in a container using Docker runtime.A maximum of 1152 huge pages will be used if available`
```
Comment (vitalykorolev) [inline README.md:1091]: Is that an actual log message? It's missing a space after the first sentence.

**Review by vitalykorolev — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -844,174 +908,201 @@ $ docker run -d -it -p 7000:8000 -p 7001:8001 -p 7002:8002 \
...
+
+## Container Runtime Detection
+
+Since MarkLogic 11.2, MarkLogic is able to detect on which container runtime it is running on.
+
+### Docker Engine
+
+When running on Docker Engine the following entry will show up in the ErrorLogs.txt:
+
+`2024-03-15 08:27:36.136 Info: MarkLogic Server is running in a container using Docker runtime. A maximum of 1152 huge pages will be used if available`
```
Comment (jkerr5) [inline README.md:1091]: Change the log message examples to 
`Info: MarkLogic Server is running in a container using Docker runtime. A maximum of <x> huge pages will be used if available`

File: README.md [hunk 5]
```diff
@@ -844,174 +908,201 @@ $ docker run -d -it -p 7000:8000 -p 7001:8001 -p 7002:8002 \
...
 
 This tag points to the latest minor, patch, and major release of MarkLogic Server on CentOS. The image will contain the latest supporting scripts and OS patches.
 
 It will pull the latest image and can cross patch, minor or major release numbers (11.0.0, 11.0.1, 11.1.0, 11.1.1, 12.0.0, etc.)
 
+**Note: The 'latest' images should not be used in production.**
+
+## Container Runtime Detection
+
+Since MarkLogic 11.2, MarkLogic is able to detect on which container runtime it is running on.
```
Comment (jkerr5) [inline README.md:1085]: Starting with version 11.2, MarkLogic is able to detect which container runtime the container is running in.

File: README.md (hunk 4 above)
Comment (jkerr5) [reply README.md:1091]: Add

"Where <x> is calculated as described in the [Configuring HugePages](configuring-hugepages) section."

Comment (jkerr5) [reply README.md:1091]: Do this for each of the container type sections.

**Review by jkerr5 — CHANGES_REQUESTED** [review]

**Review by jkerr5 — APPROVED** [review]

**Review by vitalykorolev — APPROVED** [review]

**Pull request merged**